p2p/src/peer_manager/mod.rs

```diff
@@ -74,19 +77,44 @@ use self::{
     dns_seed::{DefaultDnsSeed, DnsSeed},
     peer_context::{PeerContext, SentPing},
     peerdb::storage::PeerDbStorage,
+    peers_eviction::{
+        PreservedInboundCountAddressGroup, PreservedInboundCountNewBlocks,
+        PreservedInboundCountNewTransactions, PreservedInboundCountPing,
+    },
 };
 
-/// Desired number of full relay outbound connections.
-/// This value is constant because users should not change this.
-pub const OUTBOUND_FULL_RELAY_COUNT: usize = 8;
+#[derive(Default, Debug)]
+pub struct ConnectionCountLimits {
+    pub preserved_inbound_count_address_group: PreservedInboundCountAddressGroup,
+    pub preserved_inbound_count_ping: PreservedInboundCountPing,
+    pub preserved_inbound_count_new_blocks: PreservedInboundCountNewBlocks,
+    pub preserved_inbound_count_new_transactions: PreservedInboundCountNewTransactions,
```

Comment:
FYI: I ended up not needing to modify the inbound connection limits in the tests. But I decided to keep them here anyway, near their outbound counterparts, for completeness and because they may become useful in the future.